pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -15,101 +15,29 @@ package aws
 import (
 	"context"
 	"fmt"
-	"time"
```

Comment:
I'm not sure I'm sold on the restructuring. Why did we make this a single file package and then move all the othe providers into a folder? Seems simpler to me to keep them all at the top level

Reply:
I think you're right that moving it hasn't really cleaned up anything and maybe made some things worse like having to export the fields for the suite tests to manipulate. My initial thought on breaking this out was to keep these minimal because there was logic building up in these files which complicates the simple provider abstraction. I also wanted to limit the exposed providers from all of them, to just the ones directly used by the CP interface. For example, the AMIProvider and SecurityGroupsProvider are only depended on by the LaunchTemplate Provider.

Reply:
> Minimally, it makes the diff somewhat hard to review

+1 I'm having a hard time trying to follow all the changes here too. However, I do like the AMIFamily interface and the separation we've got there.

Reply:
moved providers back to a flat layout, should improve the diff